cli/lsp/testing/server.rs

```diff
@@ -87,6 +89,7 @@ impl TestServer {
           match update_rx.recv().await {
             None => break,
             Some(snapshot) => {
+              let config = &language_server.0.read().await.config;
```

Comment:
We should probably use a `snapshot.config (ConfigSnapshot)` here instead of locking the language server.